src/test/java/com/puppycrawl/tools/checkstyle/LocalizedMessageTest.java

```diff
@@ -0,0 +1,230 @@
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+import java.util.Locale;
+import java.util.ResourceBundle;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.LocalizedMessage.Utf8Control;
+
+/**
+ * Custom class loader is needed to pass URLs to pretend these are loaded from the classpath
+ * though we can't add/change the files for testing. The class loader is nested in this class,
+ * so the custom class loader we are using is safe.
+ *
+ * @noinspection ClassLoaderInstantiation
+ * @noinspectionreason ClassLoaderInstantiation - Custom class loader is needed to
+ *      pass URLs for testing
+ */
+public class LocalizedMessageTest {
+
+    @Test
+    public void testNullArgs() {
+        final LocalizedMessage messageClass = new LocalizedMessage(Definitions.CHECKSTYLE_BUNDLE,
+                Locale.getDefault(), DefaultLogger.class, "DefaultLogger.addException", "myfile");
+        assertWithMessage("Violation should contain exception info")
+                .that(messageClass.getMessage())
+                .contains("Error auditing myfile");
+
+        final LocalizedMessage nullClass = new LocalizedMessage(Definitions.CHECKSTYLE_BUNDLE,
+                Locale.getDefault(), DefaultLogger.class, "DefaultLogger.addException");
+        final String outputForNullArgs = nullClass.getMessage();
+        assertWithMessage("Violation should contain exception info")
+                .that(outputForNullArgs)
+                .contains("Error auditing {0}");
+    }
+
+    @Test
+    public void testBundleReloadUrlNull() throws IOException {
+        final Utf8Control control = new Utf8Control();
+        final ResourceBundle bundle = control.newBundle(
+                "com.puppycrawl.tools.checkstyle.checks.coding.messages",
+                Locale.ENGLISH, "java.class",
+                Thread.currentThread().getContextClassLoader(), true);
+        assertWithMessage("Bundle should be null when reload is true and URL is null")
+            .that(bundle)
+            .isNull();
```

Comment:
Indentation +4

Reply:
Curious why as this is the exact same as where it was moved from.
https://github.com/checkstyle/checkstyle/pull/12102/files#diff-de1cc4a0122ed19141555104a66300e79271973240674019befb55f2f8ac76faL138

Reply:
Then at least pick one, other wrapped method calls in this PR are +8, this one is only +4 :) But wrapped method calls should be +8 I think.

Reply:
Technically I am not picking anything as the code is copied/pasted from the original location so all LocalizedMessage tests are in one place.

Reply:
This and the others are done.